examples/2d/cifar_torch.py

```diff
@@ -165,11 +165,12 @@ def test(model, device, test_loader, scattering):
 
     # Optimizer
     lr = 0.1
+    optimizer = torch.optim.SGD(model.parameters(), lr=lr, momentum=0.9,
+                                        weight_decay=0.0005)
+    scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=20, gamma=0.2)
     for epoch in range(0, 90):
         if epoch%20==0:
-            optimizer = torch.optim.SGD(model.parameters(), lr=lr, momentum=0.9,
-                                        weight_decay=0.0005)
-            lr*=0.2
+            scheduler.step()
```

Comment:
Since you are using the `StepLR` learning rate scheduler, this step should be done at the end of every update.